File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/core/Robot.java
package org.firstinspires.ftc.teamcode.core;

import com.qualcomm.robotcore.hardware.HardwareMap;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

public abstract class Robot {
    protected List<Component> components = new ArrayList<>();

    public void addComponents(Component... components){
        this.components.addAll(Arrays.asList(components));
    }

    protected abstract void mapHardware(HardwareMap hardwareMap);
}

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/core/Robot.kt
package org.firstinspires.ftc.teamcode.core;

import com.qualcomm.robotcore.hardware.HardwareMap;

import java.util.ArrayList;
import java.util.Arrays;

abstract class Robot {
    val components = mutableListOf<Component>()

    fun addComponents(vararg components: Component){
        this.components.addAll(components);
    }

    abstract fun mapHardware(map: HardwareMap);
}
